### PR TITLE
[14.0][IMP] shopfloor zone picking improve select_line endpoint/state

### PR DESCRIPTION
**Pull request by TDu**
ref.: rau-214 rau-215

### CODE REVIEW

File: shopfloor_mobile/static/wms/src/scenario/zone_picking.js [hunk 1]
```diff
@@ -613,15 +625,33 @@ const ZonePicking = {
...
                                 this.state.data.scan_location_or_pack_first &&
-                                !sublocation
+                                !sublocation &&
+                                !pack
                             ) {
+                                if (this.all_lines_with_package()) {
+                                    return "Scan pack";
+                                }
                                 return "Scan location / pack";
                             }
```
Comment (jbaudoux) [inline shopfloor_mobile/static/wms/src/scenario/zone_picking.js]:
```
                            if (this.state.data.scan_location_or_pack_first)
                                if (!pack && this.all_lines_with_package() {
                                    return "Scan pack";
                                }
                                if (!sublocation && !pack) {
                                    return "Scan location / pack";
                                }
                                if (any_line_with_lot) {
                                    return "Scan product / lot";
                                } else {
                                    return "Scan product";
                                }
                            }
```

File: shopfloor_mobile/static/wms/src/scenario/zone_picking.js [hunk 2]
```diff
@@ -613,15 +625,33 @@ const ZonePicking = {
...
+                                } else {
+                                    return "Scan product";
+                                }
+                            }
                             if (sublocation) {
                                 return "Scan product / lot / package";
                             }
+                            if (pack) {
+                                return "Scan product / lot / location";
+                            }
```
Comment (jbaudoux) [inline shopfloor_mobile/static/wms/src/scenario/zone_picking.js]:
```
if sublocation
  if pack
    if any_line_with_lot
      scan product / lot
    scan product
  else
    if any_line_with_lot
      scan pack / product / lot
    scan pack /product
else
  if any_line_with_lot
    scan location / pack / product / lot
  scan location / pack /product
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]

File: shopfloor_mobile/static/wms/src/scenario/zone_picking.js [hunk 3]
```diff
@@ -613,16 +637,36 @@ const ZonePicking = {
...
+                                    if (this.any_line_with_lot()) {
+                                        return "Scan product / lot";
+                                    }
+                                    return "Scan product";
+                                } else {
+                                    if (this.any_line_with_lot()) {
+                                        return "Scan product / pack / lot";
+                                    }
+                                    return "Scan product / pack";
+                                }
```
Comment (jbaudoux) [inline shopfloor_mobile/static/wms/src/scenario/zone_picking.js]: pack / product / lot
Why having pack after product? Or product / lot / pack


pack / product ?

Comment (TDu) [reply shopfloor_mobile/static/wms/src/scenario/zone_picking.js]: changed

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
@TDu Can you squash the fixups?

**Comment by TDu** [comment]
squashed

**Comment by jbaudoux** [comment]
rebased

**Review by mmequignon — APPROVED** [review]

**Pull request merged**